go/tasks/plugins/webapi/databricks/plugin.go

```diff
@@ -0,0 +1,316 @@
+}
+
+func (p Plugin) ResourceRequirements(_ context.Context, _ webapi.TaskExecutionContextReader) (
+	namespace core.ResourceNamespace, constraints core.ResourceConstraintsSpec, err error) {
+
+	// Resource requirements are assumed to be the same.
+	return "default", p.cfg.ResourceConstraints, nil
+}
+
+func (p Plugin) Create(ctx context.Context, taskCtx webapi.TaskExecutionContextReader) (webapi.ResourceMeta,
+	webapi.Resource, error) {
+	taskTemplate, err := taskCtx.TaskReader().Read(ctx)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	token, err := taskCtx.SecretManager().Get(ctx, p.cfg.TokenKey)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	container := taskTemplate.GetContainer()
+	sparkJob := plugins.SparkJob{}
+	err = utils.UnmarshalStruct(taskTemplate.GetCustom(), &sparkJob)
+	if err != nil {
+		return nil, nil, errors.Wrapf(pluginErrors.BadTaskSpecification, err, "invalid TaskSpecification [%v], failed to unmarshal", taskTemplate.GetCustom())
+	}
+
+	modifiedArgs, err := template.Render(ctx, container.GetArgs(), template.Parameters{
+		TaskExecMetadata: taskCtx.TaskExecutionMetadata(),
+		Inputs:           taskCtx.InputReader(),
+		OutputPath:       taskCtx.OutputWriter(),
+		Task:             taskCtx.TaskReader(),
+	})
+	if err != nil {
+		return nil, nil, err
+	}
+
+	decodeBytes, err := base64.StdEncoding.DecodeString(sparkJob.DatabricksConf)
+	if err != nil {
+		return nil, nil, fmt.Errorf("failed to decode databricksJob: %v: %v", sparkJob.DatabricksConf, err)
+	}
+
+	databricksJob := make(map[string]interface{})
+	err = json.Unmarshal(decodeBytes, &databricksJob)
+	if err != nil {
+		return nil, nil, fmt.Errorf("failed to unmarshal databricksJob: %v: %v", decodeBytes, err)
+	}
+
+	if err != nil {
+		return nil, nil, err
+	}
+
+	databricksJob["spark_python_task"] = map[string]interface{}{"python_file": p.cfg.EntrypointFile, "parameters": modifiedArgs}
+
+	req, err := buildRequest(post, databricksJob, p.cfg.databricksEndpoint,
+		p.cfg.DatabricksInstance, token, "", false)
+	if err != nil {
+		return nil, nil, err
+	}
+	resp, err := p.client.Do(req)
+	if err != nil {
+		return nil, nil, err
+	}
+	defer resp.Body.Close()
+	data, err := buildResponse(resp)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	if data["run_id"] == "" {
+		return nil, nil, pluginErrors.Wrapf(pluginErrors.RuntimeFailure, err,
+			"Unable to fetch statementHandle from http response")
+	}
+	runID := fmt.Sprintf("%v", data["run_id"])
+
+	return &ResourceMetaWrapper{runID, p.cfg.DatabricksInstance, token},
+		&ResourceWrapper{StatusCode: resp.StatusCode}, nil
+}
+
+func (p Plugin) Get(ctx context.Context, taskCtx webapi.GetContext) (latest webapi.Resource, err error) {
+	exec := taskCtx.ResourceMeta().(*ResourceMetaWrapper)
+	req, err := buildRequest(get, nil, p.cfg.databricksEndpoint,
+		p.cfg.DatabricksInstance, exec.Token, exec.RunID, false)
+	if err != nil {
+		logger.Info(ctx, "Failed to build databricks job request [%v]", err)
```

Comment:
Should it be logged as an error instead of info?

Reply:
updated, thanks